actioncable/lib/action_cable/connection/web_socket_rack.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+module ActionCable
+  module Connection
+    # Wrap the real socket to minimize the externally-presented API
+    #
+    # This variation leverages the `rack.upgrade?` approach detailed here: https://github.com/rack/rack/pull/1272
+    class WebSocketRack # :nodoc:
+      UPGRADE_EXISTS = "rack.upgrade?".freeze
+      UPGRADE = "rack.upgrade".freeze
+      PROTOCOL_NAME_IN = "HTTP_SEC_WEBSOCKET_PROTOCOL".freeze
+      PROTOCOL_NAME_OUT = "Sec-Websocket-Protocol".freeze
+      CLOSE_REASON = "".freeze
+      def initialize(env, event_target, event_loop, protocols)
+        env[UPGRADE] = self
+        @event_target = event_target
+        @protocol = nil
+        @websocket = nil
+        if env[PROTOCOL_NAME_IN]
+          if env[PROTOCOL_NAME_IN].is_a?(String)   # for single list headers such as "json, soap, foo"
+            env[PROTOCOL_NAME_IN].split(/,[\s]?/).each { |i| next unless protocols.include?(i); @protocol = i; break; }
```

Comment:
Not a big fan of the semi-colon separated code, prefer to see this kind of thing broken out by newlines.

```ruby
loop.each do |i|
  line_1
  line_2
  line_3
end
```

Reply:
Sure, I can fix this.